src/util/tr.js

```diff
@@ -877,6 +943,66 @@ const _tr = (msg, {
     })
   }
 
+  const triggerByInactivity = (targetNotif) => {
+    const IDLE_TIME_THRESHOLD = targetNotif.display.deliveryTrigger.inactive * 1000 // 5 seconds
+    let idleTimer
+
+    function removeEventListeners () {
+      window.removeEventListener('mousemove', resetIdleTimer)
+      window.removeEventListener('keypress', resetIdleTimer)
+      window.removeEventListener('scroll', resetIdleTimer)
+      window.removeEventListener('mousedown', resetIdleTimer)
+      window.removeEventListener('touchmove', resetIdleTimer)
+      window.removeEventListener('click', resetIdleTimer)
+    }
+
+    window.addEventListener('mousemove', resetIdleTimer)
+    window.addEventListener('keypress', resetIdleTimer)
+    window.addEventListener('scroll', resetIdleTimer)
+    window.addEventListener('mousedown', resetIdleTimer)
+    window.addEventListener('touchmove', resetIdleTimer)
+    window.addEventListener('click', resetIdleTimer)
+
+    resetIdleTimer()
+
+    // Function to reset the idle timer
+    function resetIdleTimer () {
+      clearTimeout(idleTimer)
+      idleTimer = setTimeout(function () {
+        renderFooterNotification(targetNotif)
+        removeEventListeners()
+        // Perform actions here when user is idle for 10 seconds
+      }, IDLE_TIME_THRESHOLD)
+    }
+  }
+
+  const triggerByScroll = (targetNotif) => {
+    function calculateScrollPercentage () {
+      const fullHeight = document.documentElement.scrollHeight
+
+      const scrollPosition = window.scrollY || window.pageYOffset || document.documentElement.scrollTop
+
+      const scrollPercentage = (scrollPosition / (fullHeight - window.innerHeight)) * 100
+
+      return scrollPercentage
+    }
```

Comment:
Why are extra lines between each line of code in this function

Reply:
Done